File: 169. Majority Element.py
class Solution(object):
    def majorityElement(self, nums):
        n= len(nums)
        eMap = {}
        for i in range(n):
            if nums[i] not in eMap:
                eMap[nums[i]] = 0
            eMap[nums[i]] += 1
            if  eMap[nums[i]] > n//2:
                return nums[i]

        

File: 169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int freq = 1;
        int majority = nums[0];
        for(int i=1; i<nums.length; i++){
            if (nums[i] == majority){
                freq += 1;
            }else if (freq > 0){
                freq -= 1;
            } else{
                freq += 1;
                majority = nums[i];  
            }
        }
        return majority;
    }
}